libqtile/window.py

```diff
@@ -727,12 +727,23 @@ def update_strut(self):
             "_NET_WM_STRUT_PARTIAL",
             unpack=int
         )
-        strut = strut or self.window.get_property(
-            "_NET_WM_STRUT",
-            unpack=int
-        )
         if strut:
-            self.qtile.add_strut(strut)
+            x_screen_dimensions = self.qtile.root.get_geometry()
+            if strut[0]:    # left
+                x = abs(strut[0])
+                y = abs((strut[4] + strut[5]) / 2 or (strut[6] + strut[7]) / 2)
+            elif strut[1]:  # right
+                x = x_screen_dimensions.width - abs(strut[1])
+                y = abs((strut[4] + strut[5]) / 2 or (strut[6] + strut[7]) / 2)
+            elif strut[2]:  # top
+                x = abs((strut[8] + strut[9]) / 2 or (strut[10] + strut[11]) / 2)
+                y = abs(strut[2])
+            else:           # bottom
+                x = abs((strut[8] + strut[9]) / 2 or (strut[10] + strut[11]) / 2)
```

Comment:
What happens when a window sets the property incorrectly and doesn't provide enough values here? I'm guessing that the `IndexError` is caught by the eventloop, logged and qtile continues on as before. Do you know?

Reply:
I just forced an `IndexError` to occur and for me that results in qtile continuing, obviously without the static window, and logging the exception.

Reply:
Ah, nice.